components/BarcodeScanner.vue

```diff
@@ -138,24 +152,54 @@ export default {
       cameraConstraints.zoom.step = step;
     }
 
+    function startBarcodeDetection() {
+      // Get video element
+      const videoElement = document.querySelector(
+        '#dce-video-container > video'
+      );
+      const {BarcodeDetector} = globalThis;
+      // Check if BarcodeDetector is supported
+      if(!BarcodeDetector) {
+        alert('Barcode Detector is not supported in this browser.');
+        return;
+      } else {
+        const barcodeDetector = new BarcodeDetector({formats: ['pdf417']});
+        const detectBarcode = async video => {
+          try {
+            // Detect barcodes in the current video frame
+            const barcodes = await barcodeDetector.detect(video);
+            if(barcodes.length > 0) {
+              barcodes.forEach(barcode => {
+                const {format, rawValue} = barcode;
+                if(format && rawValue) {
+                  console.log('Barcode Detection API:', barcode);
+                  emit('result', {type: format, text: rawValue});
+                }
+              });
+            }
+          } catch(error) {
+            console.error('Barcode detection failed:', error);
+          }
+          // Schedule the next frame check
+          video.requestVideoFrameCallback(() => detectBarcode(video));
+        };
+        // Start the detection loop
+        videoElement.requestVideoFrameCallback(() =>
+          detectBarcode(videoElement)
+        );
```

Comment:
Style nit ... but also, should this say `video` or `videoElement`? It's not consistent with the above call.

```suggestion
        videoElement.requestVideoFrameCallback(
          () => detectBarcode(video));
```

Reply:
Updated in 6b621b2